### PR TITLE
React router

### PR DESCRIPTION
**Pull request by joshua-fogg**
added in React Router
add message list route /
add wildcard backend
add link to /new

### CODE REVIEW

File: client/components/LoadMessages.jsx [hunk 1]
```diff
@@ -1,11 +1,11 @@
 import React from 'react'
-import {connect} from 'react-redux'
-import {fetchMessages} from '../actions'
+import { connect } from 'react-redux'
+import { fetchMessages } from '../actions'
```
Comment (don-smith) [inline client/components/LoadMessages.jsx]: This kind of spacing isn't consistent with the rest of the project. Is this the doing of your linter?

File: client/components/LoadMessages.jsx [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
 
-function LoadMessages ({dispatch}) {
-    return ( <button onClick={() => dispatch(fetchMessages())}>
-             Fetch Messages
+function LoadMessages ({ dispatch }) {
+  return (<button onClick={() => dispatch(fetchMessages())}>
+    Fetch Messages
              </button>
```
Comment (don-smith) [inline client/components/LoadMessages.jsx]: This button looks banged up.

File: client/components/App.jsx [hunk 3]
```diff
@@ -1,16 +1,18 @@
...
 export default class App extends React.Component {
   render () {
     return (
       <div className='app'>
+        <Route exact path ='/' component={MessageList}/>
+        <Link to='/new'> Create New Message </Link>
         <LoadMessages/>
-        <MessageList/>
-        <NewMessage />
+        <Route path='/new' component={NewMessage} />
```
Comment (don-smith) [inline client/components/App.jsx]: Move this up

File: client/index.js [hunk 4]
```diff
@@ -15,7 +16,9 @@ let store = createStore(reducers, compose(
 document.addEventListener('DOMContentLoaded', () => {
   render(
     <Provider store={store}>
+      <Router>
       <App />
```
Comment (don-smith) [inline client/index.js]: Indent

File: server/server.js [hunk 5]
```diff
@@ -7,5 +7,8 @@ var server = express()
 server.use(express.static(path.join(__dirname, '../public')))
 
 server.use('/api/v1/', apiRoutes)
+server.get('*', (req, res) => {
+    res.sendFile(path.join(__dirname, '../public/index.html'))
```
Comment (don-smith) [inline server/server.js]: de-indent

**Comment by joshua-fogg** [comment]
tweaks completed , please review

**Pull request merged**